### PR TITLE
Adding transportType to ranger. And building ServiceNode with transportType

### PR DESCRIPTION
**Pull request by koushikr**
1. Added a portScheme to ServiceNode 
2. Added the same to the builders

### CODE REVIEW

File: ranger-core/src/main/java/io/appform/ranger/core/model/ServiceNode.java [hunk 1]
```diff
@@ -30,8 +30,13 @@
     private String host;
     private int port;
     private T nodeData;
+    @Builder.Default
     private HealthcheckStatus healthcheckStatus = HealthcheckStatus.healthy;
+    @Builder.Default
     private long lastUpdatedTimeStamp = Long.MIN_VALUE;
+    //Can be any scheme that you intend to init.
+    @Builder.Default
+    private String scheme = "http";
```
Comment (santanusinha) [inline ranger-core/src/main/java/io/appform/ranger/core/model/ServiceNode.java:39]: * rename this to `portScheme`.
* create a class called `PortSchemes` that has string constants like HTTP, HTTPS, TCP etc with values "http", "https", "tcp" respectively
* set portScheme deffault to portScheme = PortSchemes.HTTP

File: ranger-core/src/main/java/io/appform/ranger/core/serviceprovider/BaseServiceProviderBuilder.java [hunk 2]
```diff
@@ -51,6 +51,7 @@
     protected S serializer;
     protected String hostname;
     protected int port;
+    protected String scheme;
```
Comment (santanusinha) [inline ranger-core/src/main/java/io/appform/ranger/core/serviceprovider/BaseServiceProviderBuilder.java:54]: portScheme

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+## [1.0-RC10]
+- Introduced a port_scheme on ServiceNode, so multiple types of protocols could be supported. Was assumed to be HTTP instead.
```
Comment (santanusinha) [inline CHANGELOG.md:5]: `port_scheme` -> `portScheme`

File: ranger-core/src/main/java/io/appform/ranger/core/serviceprovider/BaseServiceProviderBuilder.java [hunk 4]
```diff
@@ -199,7 +205,8 @@ protected final ServiceProvider<T, S> buildProvider() {
                 .add(healthcheckUpdateSignalGenerator)
                 .addAll(additionalRefreshSignals)
                 .build();
-        val serviceProvider = new ServiceProvider<>(service, ServiceNode.<T>builder().host(hostname).port(port).nodeData(nodeData).build(),
+        val serviceNode = ServiceNode.<T>builder().host(hostname).port(port).scheme(scheme).nodeData(nodeData).build();
```
Comment (santanusinha) [inline ranger-core/src/main/java/io/appform/ranger/core/serviceprovider/BaseServiceProviderBuilder.java:208]: Put chain functions on different lines

File: ranger-http/src/test/java/io/appform/ranger/http/serviceprovider/HttpShardedServiceProviderBuilderTest.java [hunk 5]
```diff
@@ -53,6 +53,7 @@ public TestNodeData(@JsonProperty("farmId") String farmId) {
     public void testProvider() throws Exception {
         val farmNodeData = TestNodeData.builder().farmId("farm1").build();
         val testNode = ServiceNode.<TestNodeData>builder().host("127.0.0.1").port(80).nodeData(farmNodeData).build();
+        Assert.assertEquals("http", testNode.getScheme());
```
Comment (santanusinha) [inline ranger-http/src/test/java/io/appform/ranger/http/serviceprovider/HttpShardedServiceProviderBuilderTest.java:56]: `Assert.assertEquals(PortSchemes.HTTP, testNode.getScheme());`

**Review by santanusinha — CHANGES_REQUESTED** [review]

File: ranger-core/src/main/java/io/appform/ranger/core/model/ServiceNode.java (hunk 1 above)
Comment (koushikr) [reply ranger-core/src/main/java/io/appform/ranger/core/model/ServiceNode.java]: Booted with the same constants. Resolving this!

**Review by santanusinha — APPROVED** [review]

**Pull request merged**